tests/test_timeout_service.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 MOSEC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import random
+import shlex
+import socket
+import subprocess
+import time
+from http import HTTPStatus
+from typing import Any, Type
+
+import httpx
+import pytest
+
+from mosec import Worker
+
+
+def wait_for_port_open(host: str, port: int, timeout: int):
+    start_time = time.monotonic()
+    while time.monotonic() - start_time < timeout:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            sock.connect((host, port))
+            sock.shutdown(socket.SHUT_RDWR)
+            return True
+        except (ConnectionRefusedError, OSError):
+            pass
+        finally:
+            sock.close()
+        time.sleep(0.1)
+    return False
+
+
+def wait_for_port_free(host: str, port: int, timeout: int):
```

Comment:
Never used?

Reply:
I realized it's not necessary to wait for the port to be available but forget to remove it.